internal/support/scheduler/rest_interval.go

```diff
@@ -254,23 +263,30 @@ func restDeleteIntervalByName(
 	r *http.Request,
 	lc logger.LoggingClient,
 	dbClient interfaces.DBClient,
-	scClient interfaces.SchedulerQueueClient) {
+	scClient interfaces.SchedulerQueueClient,
+	intervalActionLoader interval.IntervalActionLoader,
+	intervalLoader interval.IntervalLoader) {
```

Comment:
I think this comment still applies. Can we look into this? If this is valid I think we can also applies this to the operator structs which required both of these constructs separately.

Reply:
I apologize for having missed replying (earlier on) that I had looked into this, and while it may well be possible to combine those two arguments, condensing them into one, by using the type `IntervalDeleter`, the ripple from this is significant. This applies to both (production) code as well as, and especially, and even more so, to the numerous unit tests which would require refactoring. In that light, I _highly_ recommend that we don't undertake such refactorings at this time.

Reply:
OK, I will create a new issue calling out this tech debt.

Reply:
After taking a second pass at this, I can see that the arguments have duplicate functionality. For example:
https://github.com/edgexfoundry/edgex-go/blob/530c35f1cd71203c25442055d8d029b8fc27e873/internal/support/scheduler/operators/interval/delete.go#L146-L148

Here we are passing `IntervalActionLoader`,`IntervalLoader`, and `IntervalDeleter` to the same function where `IntervalDeleter` encapsulates both `IntervalActionLoader` and`IntervalLoader`. Furthermore, in that example we are not even using `IntervalActionLoader` nor`IntervalLoader` but rather only `IntervalDeleter` . I think this is confusing and significant enough where I do not think we should introduce this into the codebase to be handled later.  If this was not as prominent and less confusing then I would agree to a follow up tech debt issue.

As for the amount of effort to refactor and trying to avoid it, I think that really does not apply here since this PR is the one introducing all the tech debt. In this PR we are fixing a bug, which means introducing some missed functionality, which requires changes from the REST handler(s) up the call stack to the operator(s). So really there is no real refactoring that needs to be done, we simply need to make the changes in this PR fit better with the intended solution rather than taking the easiest route to get the functionality in.

Reply:
- The point above about the overlap between a number of related struct types (`IntervalActionLoader`, `IntervalLoader`, and `IntervalDeleter`, etc.) is well-taken 👍
- I agree and will address the removal of this overlap as referenced above.

Reply:
The ask above has been addressed and a review of this `PR` requested 👍